### PR TITLE
Added attributes and comments to generated code

### PR DESCRIPTION
**Pull request by teddybeermaniac**
Hi, after setting resource's AccessModifier to Public in my project, I noticed many warnings about missing comments for publicly visible types, or members, so I modified your code to add those comments based on what original resx generator does.
I also added the same attributes that it adds to the class.

### CODE REVIEW

**Comment by MarkPflug** [comment]
Thanks so much for the contribution. This is a great addition.

There are a couple issues I notice though.

Consider the following values:

```
"Strings" {
"NewLineInString": "Hello
World",
"NewLineWithCommentBreaker":"Hello
/World",
}
```

These values, when processed into comments would look like:

```
/// <summary> Looks up a localized string similar to Hello
world
/// </summary>
/// <summary> Looks up a localized string similar to Hello
/world
/// </summary>
```

The obvious solution to fixing the first issue, is to put `///` at the start of every new line. In doing so, the second example will still be broken, but more subtly:
```
/// <summary> Looks up a localized string similar to Hello
///world
/// </summary>
/// <summary> Looks up a localized string similar to Hello
////world
/// </summary>
```

The extra '/' breaks the C# triple slash code-comment, and if you compile with DocumentationFile enabled and TreatWarningsAsErrors this code will fail to compile. So, adding a `/// ` (note the space) should fix both cases.

I added unit tests for this in my master branch. I also fixed the unit tests (on my machine). The tests tend to break every once in a while because they use external MSBuild environment to do their work. Let me know if you have any luck running them on your machine.

I'll point out as well, the second example is broken in .resx files to this day, so if you fix that you'll be doing better than the guys at Microsoft! :)

**Comment by MarkPflug** [comment]
It also might be good to truncate the value for inclusion in the comment. 100 characters should be enough to figure out what string will be returned.

**Comment by teddybeermaniac** [comment]
Hi! Thanks for the comments, I should learn that making small changes, just to cover the case that I need might not always be enough. :smile:

Yeah, now I got the tests to run, even with the added difficulty of doing so on Linux, so I'm going to do some fixes now.

File: Elemental.JsonResource/JsonResourceGenerator.cs [hunk 1]
```diff
@@ -153,71 +157,111 @@ public override bool Execute()
 						{
 							if (!string.IsNullOrEmpty(ns))
 							{
-								w.WriteLine("namespace " + ns + " {");
+								w.WriteLine($"namespace {ns} {{\n");
```
Comment (MarkPflug) [inline Elemental.JsonResource/JsonResourceGenerator.cs:160]: What purpose does the '\n' serve? This is a call two WriteLine, so it should have a newline (platform-specific) already. Using '\n' will cause IDEs to complain if someone views the files.
```suggestion
								w.WriteLine($"namespace {ns} {{");
```

File: Elemental.JsonResource/JsonResourceGenerator.cs [hunk 2]
```diff
@@ -153,71 +157,111 @@ public override bool Execute()
...
+												{
+													comment = $"Looks up a {item.Key} text file";
+												}
+
+												if (comment != null)
+												{
+													IEnumerable<string> commentLines = comment.Split(new[] { "\r", "\n", "\r\n" }, StringSplitOptions.None);
+													w.Write($@"
+    /// <summary>
+    /// {string.Join("\n    /// ", commentLines)}.
```
Comment (MarkPflug) [inline Elemental.JsonResource/JsonResourceGenerator.cs:233]:
```suggestion
    /// {string.Join(Environment.NewLine + "    /// ", commentLines)}.
```

File: Elemental.JsonResource/JsonResourceGenerator.cs [hunk 3]
```diff
@@ -153,71 +157,111 @@ public override bool Execute()
...
+										break;
 								}
 							}
 
 							w.WriteLine("}");
 
 							if (!string.IsNullOrEmpty(ns))
 							{
-								w.WriteLine("}");
+								w.WriteLine("\n}");
```
Comment (MarkPflug) [inline Elemental.JsonResource/JsonResourceGenerator.cs:264]:
```suggestion
								w.WriteLine("}");
```

**Comment by teddybeermaniac** [comment]
Ok, it looks that I made it work for the cases you provided.
I also went ahead and made a bit more :smile::
* The part which generated the code was a bit hard to read for me, and when I noticed that you have \> C# 6.0 set in the project, I decided to rewrite it using string interpolation, so that it's easier to see what is happening. I know, that the lack of indentation in those multi-line strings might not look the best, but I think it's not that bad.
* While I was at it I made the generated code pretty and formatted, if by any chance, someone would need to look at it.
* And on top of that, when I wrote the previous comment, I thought that the tests were failing because of the test cases you added. It turns out, that it was all because of me running Linux. So I made a few changes here, and there, to make tests multi-platform (except for build failing altogether, beause of MSBuildGitHash, which I had to comment out in the project file). I hope that I didn't break them on Windows, but I don't have access to any installation of it, so you'll have check it.
* Also for the sake of completness I added comments for Files resources, but I wasn't sure what to put there, so feel free to change it (line 225), if you have a better one in mind.

One thing I noticed is that it looks like you don't use line wrapping, or you do it at a big number, so in attempt to keep the code style I assumed you follow, some of my code has veeery long lines.

Ok, now I feel like I turned, what at the beginning was an afterthought, into a proper pull request. :smiley:

**Comment by MarkPflug** [comment]
Also, I really appreciate you testing/fixing the tests on unix!

**Comment by teddybeermaniac** [comment]
About that '\n'... Yeah, I changed similar thing in your code, and did the same myself. It's a stronger habit that I thought. :)

**Comment by MarkPflug** [comment]
No worries, I already merged. Minor change, I'll fix it before I release it.

**Comment by MarkPflug** [comment]
Michał,
Thanks again for your contribution here. It's really appreciated!
I just pushed version 0.4.0 to nuget, it should be listed shortly.

**Pull request merged**